### PR TITLE
CM list converted into table

### PR DESCRIPTION
**Pull request by Akarshit**
@alignedleft 
I thought it would be nice to do a PR before b11 is released
Things left to do
* Handle status tab (@benfry do you think i should make a separate layout for update panel or use many ifs to reuse the current one)
* Color-coding
* Ellipses in the contribution name. (@benfry can you please help me in this, see the line note)
* Remove threading(not needed any more)
* Fix that weird bug when python mode is selected from the list.

### CODE REVIEW

File: app/src/processing/app/contrib/ContributionListPanel.java [hunk 1]
```diff
@@ -78,51 +108,182 @@ public ContributionListPanel(ContributionTab contributionTab,
...
+          }
+        }
+        label.setIcon(icon);
+        if (isSelected) {
+          label.setBackground(Color.BLUE);
+        }
+        label.setOpaque(true);
+//        return table.getDefaultRenderer(Icon.class).getTableCellRendererComponent(table, icon, isSelected, false, row, column);
+      } else if (column == 1) {
+        JTextPane name = new JTextPane();
```
Comment (Akarshit) [inline app/src/processing/app/contrib/ContributionListPanel.java]: I can get the ellipses when this column is rendered as a string but then i wont get the contribution name in bold. To get the contribution name in bold i have to use html tags but with html the ellipses do away. So i am using JTextPane to render the name, but still am not getting the ellipses is there a easy way of getting those ellipses without loops and all.

Comment (benfry) [inline app/src/processing/app/contrib/ContributionListPanel.java]: Can't you call `setFont()` on the `JLabel`?

How does the ellipsis implementation work in `JLabel`?

**Comment by benfry** [comment]
I'd vote for a separate layout, rather than a bunch of `if` statements... The CM code is messy enough as it is.

File: app/src/processing/app/contrib/ContributionListPanel.java (hunk 1 above)
Comment (Akarshit) [inline app/src/processing/app/contrib/ContributionListPanel.java]: I don't have to set the font for the whole label just a part of it.

I will open a separate issue for this and look into it later.

**Pull request merged**